modules/tensor_mechanics/src/materials/ComputeElasticityTensorCP.C

```diff
@@ -0,0 +1,50 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+#include "ComputeElasticityTensorCP.h"
+#include "RotationTensor.h"
+
+template<>
+InputParameters validParams<ComputeElasticityTensorCP>()
+{
+  InputParameters params = validParams<ComputeElasticityTensor>();
+  params.addClassDescription("Compute an elasticity tensor for crystal plasticity.");
+  params.addParam<UserObjectName>("read_prop_user_object","The ElementReadPropertyFile GeneralUserObject to read element specific property values from file");
+  return params;
+}
+
+ComputeElasticityTensorCP::ComputeElasticityTensorCP(const InputParameters & parameters) :
+    ComputeElasticityTensor(parameters),
+    _read_prop_user_object(isParamValid("read_prop_user_object") ? & getUserObject<ElementPropertyReadFile>("read_prop_user_object") : NULL),
+    _local_Euler_angles(declareProperty<RealVectorValue>("Euler_angles")),
+    _crysrot(declareProperty<RankTwoTensor>("crysrot")),
+    _R(_Euler_angles)
+{
+}
+
+void
+ComputeElasticityTensorCP::computeQpElasticityTensor()
+{
+  //Properties assigned at the beginning of every call to material calculation
+  if ( _read_prop_user_object )
```

Comment:
Can make a separate virtual function that can be overridden to assign Euler angle values.